recipes/freeglut/all/test_package/CMakeLists.txt

```diff
@@ -1,10 +1,12 @@
 cmake_minimum_required(VERSION 3.1)
-project(test_package C)
+project(test_package LANGUAGES C)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
-
-find_package(GLUT REQUIRED)
+find_package(FreeGLUT REQUIRED CONFIG)
+find_package(OpenGL REQUIRED)
```

Comment:
```suggestion
find_package(FreeGLUT REQUIRED CONFIG)
find_package(GLUT REQUIRED MODULE)

find_package(OpenGL REQUIRED)

# Test whether variables from https://cmake.org/cmake/help/latest/module/FindGLUT.html are properly defined
set(_custom_vars
    GLUT_FOUND
    GLUT_INCLUDE_DIRS
    GLUT_LIBRARIES
)

foreach(_custom_var ${_custom_vars})
    if(DEFINED ${_custom_var})
        message(STATUS "${_custom_var}: ${${_custom_var}}")
    else()
        message(FATAL_ERROR "${_custom_var} not defined")
    endif()
endforeach()
```
I think that as the CMake file names are different, there would be no problem in calling both `find_package` for the module and the config in this _CMakeLists.txt_ so that we don't have to add `test_package_v1_module` and `test_package_v1`

Reply:
I think it would be very weird and confusing. We would also have to test link to different targets.

Reply:
Yes, you have to also do something like:

```
add_executable(${PROJECT_NAME}_module ../test_package/test_package.c)
target_link_libraries(${PROJECT_NAME}_module PRIVATE GLUT::GLUT OpenGL::GL)
```

and 

```
bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package_module")
self.run(bin_path, env="conanrun")
```

I prefer to reduce the number of test_packages and add some comments the the file, but that's ok, I think we can leave it like it is and merge this quickly for users. What I think would be very nice for future PR's is adding some comments to explain why we added those test_package_module in case someone is disconcerted by those.

Reply:
I understand, but I want to avoid side effects. Calling 2 different find_package against different module/config flavors in the same CMakeLists is unusual, and I'm not even sure it wouldn't lead to surprising UB due to how these files are designed by conan generators.

Reply:
Given that the package names are different (`GLUT` and `FreeGLUT`) and that the variable names and variable names are different, I fail to see what problems could arise. I would be a different story if one wanted to load both the module and the config for the _same_ dependency with the same name, unless the module has logic to call the config (like FindBoost.cmake does).